### PR TITLE
Refactor model registry routes and configuration

### PR DESCRIPTION
**Pull request by lucferbux**
Follow up of: [RHOAIENG-4346](https://issues.redhat.com//browse/RHOAIENG-4346)

## Description

Implements the proxy changes for model registry. This is still a work in progress continued in the effort of implementing Itsio authenticated routes.

Screenshot 2024-04-17 at 20 20 53

## How Has This Been Tested?

1. Tap into a cluster in DEV mode
2. Check that you can fetch resources with model registry

## Request review criteria:

Self checklist (all need to be checked):
- [X] The developer has manually tested the changes and verified that the changes work
- [X] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: backend/src/routes/api/service/modelregistry/index.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import httpProxy from '@fastify/http-proxy';
+import { KubeFastifyInstance } from '../../../../types';
+import { DEV_MODE } from '../../../../utils/constants';
+import { getParam, setParam } from '../../../../utils/proxy';
+
+export default async (fastify: KubeFastifyInstance): Promise<void> => {
+  if (DEV_MODE) {
```
Comment (andrewballantyne) [inline backend/src/routes/api/service/modelregistry/index.ts:7]: This is not needed if your endpoint has OAuth.

File: frontend/src/concepts/modelRegistry/const.ts [hunk 2]
```diff
@@ -1,2 +1,2 @@
-export const MODEL_REGISTRY_DEFINITION_NAME = 'modelregistry-sample'; // TODO: harcoded value, we need to dynamically fetch this
-export const MODEL_REGISTRY_DEFAULT_NAMESPACE = 'shared'; // TODO: harcoded value, we need to check which will be the default namespace
+export const MODEL_REGISTRY_DEFAULT_NAMESPACE = 'odh-model-registries';
```
Comment (andrewballantyne) [inline frontend/src/concepts/modelRegistry/const.ts:1]: Will this be managed for RHOAI somehow? I assume we need a TODO here

File: frontend/src/pages/modelRegistry/ModelRegistryCoreLoader.tsx [hunk 3]
```diff
@@ -1,10 +1,11 @@
 import * as React from 'react';
 import { Outlet } from 'react-router';
-import { MODEL_REGISTRY_DEFAULT_NAMESPACE } from '~/concepts/modelRegistry/const';
+
 import { ModelRegistryContextProvider } from '~/concepts/modelRegistry/context/ModelRegistryContext';
 
+// TODO: Parametrize this to make the route dynamic
 const ModelRegistryCoreLoader: React.FC = () => (
-  <ModelRegistryContextProvider namespace={MODEL_REGISTRY_DEFAULT_NAMESPACE}>
+  <ModelRegistryContextProvider modelRegistryName="modelregistry-sample">
```
Comment (andrewballantyne) [inline frontend/src/pages/modelRegistry/ModelRegistryCoreLoader.tsx:8]: I'm confused -- why would you pass in this value 🤔 Shouldn't the dropdown take care of this in the long run? Ie it's not a static or (in the future) a provided variable -- it's decided by the nature of the dropdown's selection

File: frontend/src/concepts/modelRegistry/const.ts (hunk 2 above)
Comment (lucferbux) [reply frontend/src/concepts/modelRegistry/const.ts:1]: Yes, this is managed by RHOAI, I'm sorry there's a TODO but it was mixed when i split up the PR, let me add it again here

File: frontend/src/pages/modelRegistry/ModelRegistryCoreLoader.tsx (hunk 3 above)
Comment (lucferbux) [reply frontend/src/pages/modelRegistry/ModelRegistryCoreLoader.tsx:8]: Yes, that's rigth, that's the logic I'm working on in the follow up pr, in which the dropdown will have a modelregistry selector and will pass the name of that model registry to the context provider, that's why I left the TODO there

File: backend/src/routes/api/service/modelregistry/index.ts (hunk 1 above)
Comment (andrewballantyne) [reply backend/src/routes/api/service/modelregistry/index.ts:7]: Nvm, Lucas notes there are still items going on here... they want it disabled for prod builds.

File: Makefile [hunk 4]
```diff
@@ -84,13 +84,15 @@ undeploy:
 
 ##################################
 
+# TODO: change the namespace and guess how to parametrize the name
```
Comment (christianvogt) [inline Makefile:87]: What exactly does this mean?

Comment (lucferbux) [reply Makefile]: heh, I forgot to delete that comment, good catch

**Pull request merged**